src/azure-cli/azure/cli/command_modules/resource/_packing_engine.py

```diff
@@ -0,0 +1,187 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import os
+import json
+from knack.util import CLIError
+from azure.cli.core.util import read_file_content
+from azure.cli.command_modules.resource.custom import _remove_comments_from_json
+from azure.cli.core.profiles import ResourceType, get_sdk
+
+
+class PackagedTemplate():  # pylint: disable=too-few-public-methods
+    def __init__(self, template, artifacts):
+        self.RootTemplate = template
+        self.Artifacts = artifacts
+
+
+class PackingContext():  # pylint: disable=too-few-public-methods
+    def __init__(self, root_template_directory):
+        self.RootTemplateDirectory = os.path.abspath(root_template_directory)
+        self.CurrentDirectory = os.path.abspath(root_template_directory)
+        self.Artifact = []
+
+
+# Packs the specified template and its referenced artifacts for use in a Template Spec.
+# :param template_file: The path to the template spec .json file.
+# :type name: str
+
+def Pack(cmd, template_file):
```

Comment:
Traditionally, function name should be lower case and we use lower camel-case naming with `-` to split.